tests/test_sanity.py

```diff
@@ -281,3 +281,35 @@ def test_container_name_when_stopped(
     collapsed_text = wait_for_gprofiler_container(container, output_collapsed)
     assert "py-spy> Stopped sampling because process exited" in container.logs().decode()
     assert f";{application_container_name};python" in collapsed_text
+
+
+@pytest.mark.parametrize("in_container", [False])
+@pytest.mark.parametrize("runtime", ["java"])
+@pytest.mark.parametrize("profiler_type", ["ap"])
+def test_profiling_provided_pids(
+    docker_client: DockerClient,
+    gprofiler_docker_image: Image,
+    output_directory: Path,
+    output_collapsed: Path,
+    runtime_specific_args: List[str],
+    profiler_flags: List[str],
+    application_factory: Callable[[], _GeneratorContextManager],
+    profiler_type: str,
+) -> None:
+    """
+    Tests that gprofiler will profile only processes provided via flag --pids
+    """
+    with application_factory() as pid1:
+        with application_factory() as pid2:
+            profiler_flags.extend(["--pids", str(pid1)])
+            container = start_gprofiler_in_container_for_one_session(
+                docker_client,
+                gprofiler_docker_image,
+                output_directory,
+                output_collapsed,
+                runtime_specific_args,
+                profiler_flags,
+            )
+            wait_for_gprofiler_container(container, output_collapsed)
+            assert f"Profiling process {pid1} with async-profiler" in container.logs().decode()
+            assert f"Profiling process {pid2} with async-profiler" not in container.logs().decode()
```

Comment:
You can add here logs checking that `Processes left after filtering` shows 1 and `selected {len(processes_to_profile)} processes to profile` shows 2.

Reply:
I changed it but I am not sure about asserting 

> selected {len(processes_to_profile)} processes to profile shows 2

I feel that this might lead to some misleading failures while testing. F.e If there is some background java process running in the environment it will result in failing the test

Reply:
You're right. It can make the test flaky.

Let's remove `assert "selected 2 processes to profile" in container.logs().decode()` and keep only `assert "processes left after filtering: 1" in container.logs().decode()` which is always correct.